arch-design/multicloud/Multi_Cloud_HA.md

```diff
@@ -0,0 +1,65 @@
+
+
+## Motivation and background
+As SODA provides a critical data management platform for both On-Prem and for Cloud based workloads, it is imperative that there is no bottleneck and the data and services be available even in case of any node going down. However, the current  architecture where all the components of SODA installed on a single node are a Single Point of Failure. In case of any failure of any of the services or the node going down, the data will not be accessible to the user. Hence, it's recommended to have the SODA installation in a High Available configuration, which can withstand failure of nodes and the data and services are accessible in case of any node or service going down. 
+
+## Non-Goals
+ - Clients access to the data and services is seamless. 
+ - The Services should scale up / down to provide data and services to users.
+ - Installation / Uninstallation of SODA provides options to install SODA Multi Cloud in standalone / HA mode. 
+
+
+## Assumptions and Constraints
+TBD
+
+## Use case View
+- High Availability : To provide High Availability to the SODA Multi Cloud Deployment.
+Scale up / down as per the need of data movement. 
+
+
+
+![Multi Cloud HA Design](resources/multicloud_ha.PNG)
+                        
+
+Here is the High Level Architecture for providing HA for SODA Multi Cloud services.  
+
+
+As seen in Figure-1, the scope of I release is to provide  High Availability for the SODA Multi Cloud Services to begin with and gradually migrate the other components over the subsequent releases. 
+With this in mind, the SODA node can serve as the  Container Orchestrator (K8s / Mesos / Docker Swarm) Master node and the Multi Cloud services be deployed on the Deployment Cluster as shown in Figure - 1.
+This is a generic architecture giving flexibility to the customer to choose any of the orchestration tools as required. For the PoC K8s is the tool of choice, however it can be replaced with the customer’s choice (after necessary tests are conducted). 
+
+*Though, this architecture would still not be fully HA , as there is still a single point of failure with the SODA Node not being fully HA. 
+To be fully HA, the  Master node should be clustered in a multi-master configuration, however due to resource restrictions , it is proposed to go with the Architecture depicted in Figure - 1. 
+
+## Development and Deployment Context : 
+
+### Installation:
+  Installation would need to take into account the requirements for the Deployment Cluster for the containers. The Cluster would need to be set up during the installation phase and the prerequisite softwares would need to be installed on the Nodes. Communication would need to be established between the Master SODA node and the cluster. 
```

Comment:
what is the redundancy level ? single failure or multiple failure ?

Reply:
This document was written with PoC perspective (keeping in mind the infrastructure requirements for development / deployment / tests ) , for the PoC , single failure will be targeted and later matured to multiple failure.